From b6113c2872f9fc626cf5527278f610184dce5ef3 Mon Sep 17 00:00:00 2001
From: Sinan Kaya <okaya@codeaurora.org>
Date: Sun, 17 Apr 2016 13:36:56 -0400
Subject: [PATCH 441/447] ACPI,PCI,IRQ: remove SCI penalize function

Git-commit: 9e5ed6d1fb87dc3ff0c2a94ed75fe82027a9f597
Patch-mainline: v4.7-rc1
References: fate#320512

Removing the SCI penalize function as the penalty is now calculated on the
fly.

Signed-off-by: Sinan Kaya <okaya@codeaurora.org>
Acked-by: Bjorn Helgaas <bhelgaas@google.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 arch/x86/kernel/acpi/boot.c |    2 --
 drivers/acpi/pci_link.c     |    4 ----
 include/linux/acpi.h        |    1 -
 3 files changed, 7 deletions(-)

--- a/arch/x86/kernel/acpi/boot.c
+++ b/arch/x86/kernel/acpi/boot.c
@@ -472,8 +472,6 @@ static void __init acpi_sci_ioapic_setup
 	else
 		mp_register_ioapic_irq(bus_irq, polarity, trigger, gsi);
 
-	acpi_penalize_sci_irq(bus_irq, trigger, polarity);
-
 	/*
 	 * stash over-ride to indicate we've been here
 	 * and for later update of acpi_gbl_FADT
--- a/drivers/acpi/pci_link.c
+++ b/drivers/acpi/pci_link.c
@@ -884,10 +884,6 @@ bool acpi_isa_irq_available(int irq)
 		    acpi_irq_get_penalty(irq) < PIRQ_PENALTY_ISA_ALWAYS);
 }
 
-void acpi_penalize_sci_irq(int irq, int trigger, int polarity)
-{
-}
-
 /*
  * Over-ride default table to reserve additional IRQs for use by ISA
  * e.g. acpi_irq_isa=5
--- a/include/linux/acpi.h
+++ b/include/linux/acpi.h
@@ -278,7 +278,6 @@ struct pci_dev;
 int acpi_pci_irq_enable (struct pci_dev *dev);
 void acpi_penalize_isa_irq(int irq, int active);
 bool acpi_isa_irq_available(int irq);
-void acpi_penalize_sci_irq(int irq, int trigger, int polarity);
 void acpi_pci_irq_disable (struct pci_dev *dev);
 
 extern int ec_read(u8 addr, u8 *val);
